@here/harp-mapview/lib/Utils.ts

```diff
@@ -1319,21 +1461,27 @@ export namespace MapViewUtils {
      *                     converting from geo to world coordinates.
      * @param object - The object to get the coordinates from.
      * @param location - The reference point.
+     * @param tiltAxis - Optional axis used to define the rotation about which the object's tilt
+     * occurs, the direction vector to the location from the camera is projected on the plane with
+     * the given angle.
      */
     export function extractTiltAngleFromLocation(
         projection: Projection,
         object: THREE.Object3D,
-        location: GeoCoordinates
+        location: GeoCoordinates | Vector3Like,
+        tiltAxis?: THREE.Vector3
     ): number {
         projection.localTangentSpace(location, {
             xAxis: tangentSpace.x,
             yAxis: tangentSpace.y,
             zAxis: tangentSpace.z,
             position: cache.vector3[0]
         });
-
         // Get point to object vector (dirVec) and compute the `tilt` as the angle with tangent Z.
-        const dirVec = cache.vector3[1].copy(object.position).sub(cache.vector3[0]);
+        const dirVec = cache.vector3[2].copy(object.position).sub(cache.vector3[0]);
+        if (tiltAxis) {
+            dirVec.projectOnPlane(tiltAxis);
```

Comment:
You're still using the normal at the rotation center to compute the tilt, that is, tangentSpace.z. You should project that on the plane as well, as you do here:
https://github.com/heremaps/harp.gl/pull/1885/files#diff-305bd59a411397d3c92eafbdc5c800c3bb6f0bb0040b12d5bfc6c411cdda288fR353-R354

Reply:
The projected rotation center isn't enough, because it will be under the earths surface, I will need to also adjust the length.

I will just project the normal to the tilt axis as well.